docs/proposals/2017-5-26 - Advanced Table.md

```diff
@@ -0,0 +1,446 @@
+
+### Credits
+
+ANS credits object, for identifying the author(s) of the table.
+
+### Subtype
+
+ANS subtype. Can be used to distinguish between different types of tables, so that clients can react accordingly.
+
+### No height/width or other display options
+
+These are intentionally left out.
+
+If adapters need to include any style or rendering details, they can use additional properties on any component of a table - table, row, cell, or content element.
+
+
+## Row
+
+A `row` contains a list of cells. It will only be usable within a table.
+
+### Cells
+
+List of cells that belong to this row. The order is assumed to be left to right.
+
+### Subtype
+
+ANS Subtype. Can be used to distinguish types of rows.
+
+## Data Cell / Header Cell
+
+There are two types of cells, `data_cell` and `header_cell`. They will only be usable within a row.
+
+Both `header_cell` and `data_cell` will contain a list of content elements. They have the same spec, except for the type.
+
+### Cell Elements
+
+A list of content elements that belong to the cell. This is restricted to elements of type `text` and `raw_html`.
+
+The order of elements is assumed to be top to bottom.
+
+### Subtype
+
+ANS Subtype. Can be used to distinguish types of cells.
+
+
+## Alternatives
+
+### Cell types
+
+The spec could potentially have one type for cells, `cell`, and use subtype to distinguish between `data` and `header` cells. This would make the schema smaller, but would also require clients check two fields on each cell. I don't have strong feelings on this either way.
+
+
+## Concerns
+
+### What happens if rows have different numbers of cells?
+
+Based on the order of elements and the assumed layout (left to right, or top to bottom), it is clear where each element should go. If the user wants to leave empty space, they would include empty elements. Otherwise, it will be assuemd that any mismatch in length will lead to space at the end of the element.
+
+
+### Why do we need multiple elements in a cell?
+
+We need to be able to represent multiple p tags within a cell. Furthermore, these p tags can include minor rendering details. While we don't want any of that rendering information to be in the schema, we do need to be able to put additional properties on each paragraph within a cell. Allowing mutiple elements within a cell makes this possible.
+
+### Why do we need both raw_html and text elements in the cell?
+
+Raw html makes it possible to include more advanced markup, such as span tags, on elements within a cell. Raw html would be used sparingly, only when text elements are not sufficient.
+
+While this currently isn't a problem, because text elements do not have restrictions, my understanding is that this will change in the future. I believe allowing raw_html in a cell will future proof this functionality, and avoid unnecessary migrations.
+
+An example from methode:  
+```
+<p><span>METHODOLOGY</span> \"MOSTLY, VALUE OVER MOMENTUM STOCKS\"</p>
+```
+
```

Comment:
If a table might contains raw html that cannot be represented as `text`, why not just render the whole table as a `raw_html` element? A table with 1 or more raw_html elements requires downstream consumers to be able to parse and render HTML to render the table in its entirety.